book/configuration.rst

```diff
@@ -1,15 +1,55 @@
 .. index::
    single: Configuration
+   single: Environments
 
-Configuring Symfony (and Environments)
-======================================
+Configuring Symfony and Environments
+====================================
+
+.. index::
+   single: Configuration; Formats
+
+Configuration Formats
+---------------------
+
+Symfony supports four configuration formats: YAML, XML and PHP. Throughout
+the chapters, all configuration examples are shown in all four formats. Each
+has its own advantages and disadvantages. The choice of which to use is up
+to you:
+
+* *YAML*: Simple, clean and readable (learn more about YAML in the Yaml
+  component documentation :doc:`/components/yaml/yaml_format`);
+
+* *XML*: More powerful than YAML at times and supports IDE autocompletion;
+
+* *PHP*: Very powerful but less readable than standard configuration formats.
+
+* *Annoatations*: Code and configuration are in the same place, simple to
+  learn and to use, concise to write.
+
+The supported formats with their orders are:
+
+* Configuration (including services): YAML, XML, PHP
+
+* Routing: Annotations, YAML, XML, PHP
+
+* Validation: Annotations, YAML, XML, PHP
+
+* Doctrine Mapping: Annotations, YAML, XML, PHP
+
+* Translation: XML, YAML, PHP
```

Comment:
I'm not completely convinced we should add this section - I hate to have extra details at the top of an article. I think we should at least remove this last part (starting with `The supported formats...`) - and maybe the description of the 4 formats above that (but I'm less sure).